scalding-jdbc-macros/src/main/scala/com/twitter/scalding/jdbc/macros/impl/handler/DateTypeHandler.scala

```diff
@@ -0,0 +1,38 @@
+package com.twitter.scalding.jdbc.macros.impl.handler
+
+import scala.language.experimental.macros
+
+import scala.reflect.macros.Context
+import scala.reflect.runtime.universe._
+import scala.util.{ Success, Failure }
+
+import com.twitter.algebird.Monoid
+import com.twitter.scalding._
+import com.twitter.bijection.macros.IsCaseClass
+import com.twitter.scalding.jdbc.{ ColumnDefinition, ColumnDefinitionProvider }
+import com.twitter.scalding.jdbc.macros._
+import com.twitter.scalding.jdbc.macros.impl.FieldName
+
+/**
+ * This class contains the core macro implementations. This is in a separate module to allow it to be in
+ * a separate compilation unit, which makes it easier to provide helper methods interfacing with macros.
+ */
+object DateTypeHandler {
+
+  def apply[T](c: Context)(implicit fieldName: FieldName,
+    defaultValue: Option[c.Expr[String]],
+    annotationInfo: List[(c.universe.Type, Option[Int])],
+    nullable: Boolean): scala.util.Try[c.Expr[ColumnDefinition]] = {
+    import c.universe._
+
+    val opts = scala.util.Try[Boolean](Monoid.sum(annotationInfo.collect {
```

Comment:
True, seems like there was likely a bug there the tests never hit. Mostly as part of cleaning up the string case i've refactored all of them so the sum code doesn't exist anymore. (Here the monoid i would have been looking for was OR. But its not clean on what would occur in the presence of a repeated annotation. New code has explicit path for that)